cpp/src/parquet/stream_reader.cc

```diff
@@ -0,0 +1,224 @@
+#include "parquet/stream_reader.h"
+
+namespace parquet {
+
+StreamReader::StreamReader(std::unique_ptr<ParquetFileReader>& File_Reader)
+    : file_reader_{File_Reader.release()} {
+  file_metadata_ = file_reader_->metadata();
+
+  auto* const schema{file_metadata_->schema()};
+  auto* const group_node{schema->group_node()};
+
+  nodes_.resize(schema->num_columns());
```

Comment:
I think it is slightly more idiomatic to use reserve and push_back below (avoid unnecessary double initialization)

Reply:
I do not think the default constructor of a shared_ptr is expensive.
I prefer the code to make it absolutely clear that the nth element in the vector references the nth column.

Reply:
OK